homeassistant/components/fritz/device_tracker.py

```diff
@@ -146,24 +146,13 @@ def mac_address(self) -> str:
     @property
     def hostname(self) -> str:
         """Return hostname of the device."""
-        return self._router.devices[self._mac].hostname
+        return self._fritzbox_tools.devices[self._mac].hostname
 
     @property
     def source_type(self) -> str:
         """Return tracker source type."""
         return SOURCE_TYPE_ROUTER
 
-    @property
```

Comment:
I think we can leave this as-is.  The base class should only be for the fritz box itself so you can drop the manual device creation.

Reply:
Why not using it for both ? Works flawlessly

Reply:
Cleaned up